src/resolve/resolved-dns-query.c

```diff
@@ -312,6 +319,30 @@ static int dns_query_candidate_setup_transactions(DnsQueryCandidate *c) {
         return r;
 }
 
+static void dns_query_accept(DnsQuery *q, DnsQueryCandidate *c);
+
+static int on_candidate_timeout(sd_event_source *s, usec_t usec, void *userdata) {
+        DnsQueryCandidate *c = userdata;
+
+        assert(s);
+        assert(c);
+
+        log_debug("Accepting incomplete query candidate after expedited timeout on partial success.");
```

Comment:
`SD_RESOLVED_NO_TXT` doesn't seem to make sense anymore because services are now resolved even when their TXT RRs are never received. It's probably an unintentional side effect.

Given that it's already on its way to the stable branch (https://github.com/systemd/systemd/pull/35689) it would have been great to mention it somewhere since it changes the meaning of the resolved public API.